Mathlib/Algebra/Lie/Submodule.lean

```diff
@@ -538,6 +539,15 @@ theorem wellFounded_of_noetherian [IsNoetherian R M] :
   RelHomClass.wellFounded f (isNoetherian_iff_wellFounded.mp inferInstance)
 #align lie_submodule.well_founded_of_noetherian LieSubmodule.wellFounded_of_noetherian
 
+theorem wellFounded_of_isArtinian [IsArtinian R M] :
+    WellFounded ((· < ·) : LieSubmodule R L M → LieSubmodule R L M → Prop) :=
+  let f :
+    ((· < ·) : LieSubmodule R L M → LieSubmodule R L M → Prop) →r
```

Comment:
Would this be more idiomatic as an Order embedding?

Reply:
Oh nice: I had just mindlessly copied the `LieSubmodule.wellFounded_of_noetherian` proof (which I believe was written before had `OrderEmbedding`).

Much better now.